### PR TITLE
Serializable InsufficientBalanceException

### PR DESCRIPTION
**Pull request by longfin**
This PR fixes `InsufficientBalanceException` to be serialized properly by .NET serialization mechanism.

And more, it also fixes a bug that `UnexpectedlyTerminatedActionExceptions` doesn't contain its properties after de-serialization.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -39,6 +39,8 @@ To be released.
  -  Added `ICryptoBackend<T>.Sign(HashDigest<T>, PrivateKey)` method.
     [[#932]]
  -  `DefaultCryptoBackend` became to `DefaultCryptoBackend<T>`.  [[#932]]
+ -  Replaced `SerializationInfoExtensions.GetValueOrDefault<T>()` to
+    `SerializationInfoExtensions.TryGetValue<T>()`.  [[#940]]
```
Comment (earlbread) [inline CHANGES.md:43]: The link to the PR should be added as well.

File: Libplanet/Action/UnexpectedlyTerminatedActionException.cs [hunk 2]
```diff
@@ -55,6 +58,30 @@ StreamingContext context
...
+            }
+
+            if (info.TryGetValue($"{nameof(Action)}_type", out string actionType))
+            {
+                var values = (Dictionary)new Codec().Decode(
+                    info.GetValue<byte[]>($"{nameof(Action)}_values")
+                );
+
+                Action = (IAction)Activator.CreateInstance(Type.GetType(actionType, true, true));
+                Action.LoadPlainValue(values);
```
Comment (earlbread) [inline Libplanet/Action/UnexpectedlyTerminatedActionException.cs:83]: How about checking `null` here?

File: CHANGES.md [hunk 3]
```diff
@@ -102,6 +104,7 @@ To be released.
     instead of the longest chain.  [[#459], [#919]]
  -  `Swarm<T>.BootstrapAsync()` method became not to throw `TimeoutException` when
     it fails to connect to all neighbors.  [[#933]]
+ -  `Currency` is now serializable.  [[#940]]
```
Comment (dahlia) [inline CHANGES.md:107]: As we've never released `Currency`, this doesn't have to be recorded.

File: CHANGES.md [hunk 4]
```diff
@@ -112,8 +115,9 @@ To be released.
     [[#909]]
  -  Fixed a bug in which pre-computed state delivery had failed when a state
     key is not an `Address` when preloading.  [[#912]]
- -  Fixed a bug where `UnexpectedlyTerminatedException` hadn't been serialized
-    with `BinaryFormatter`.  [[#913]]
+ -  Fixed a bug where `UnexpectedlyTerminatedException` and
+    `InsufficientBalanceException` hadn't been serialized with
```
Comment (dahlia) [inline CHANGES.md:119]: As we've never released `InsufficientBalanceException`, this doesn't have to be recorded.

File: Libplanet/Action/UnexpectedlyTerminatedActionException.cs (hunk 2 above)
Comment (longfin) [reply Libplanet/Action/UnexpectedlyTerminatedActionException.cs:83]: Did you mean like below?

```suggestion
                Action?.LoadPlainValue(values);
```

Comment (earlbread) [reply Libplanet/Action/UnexpectedlyTerminatedActionException.cs:83]: Yeah, I thought about handling it like that or throwing an exception.

**Pull request merged**